Adding ZCTA with deprivation index

nobel1154 opened the pull request.

nobel1154 committed:
Deprivation index by zip code tabulation area

nobel1154 committed:
Update deprivation_index_by_zipcode.r

nobel1154 committed:
Merge pull request #1 from nobel1154/nobel1154-patch-1

Deprivation index by zip code tabulation area

cole-brokamp commented:
This is looking really good!, just a couple of tweaks:

- use the existing `ACS_deprivation_index_by_census_tracts.rds` within the repository rather than downloading it from github
- instead of removing the missing deprivation index values before the summarize, use `na.rm = TRUE` in the `mean(...)` function within the summarize command
- save the final `d` object as both a csv and rds file called `ACS_deprivation_index_by_zipcode.{rds,csv}` and commit these so that they will be in the repository when your pull request is merged

nobel1154 committed:
Update deprivation_index_by_zipcode.r

nobel1154 commented:
I have those changes as you described

nobel1154 committed:
test commit

nobel1154 committed:
including csv and rds files for zcta data

cole-brokamp commented:
- remove `testing_file.txt`
- cleaner code (delete commented out code and avoid overwriting R objects with the same name)
- `GEOID` should already be a character and not need to be transformed into one

cole-brokamp commented:
- include all variables making up the dep index in the zcta datasets
- don't write row numbers to CSV files

nobel1154 committed:
Clean up the Code

cole-brokamp commented on deprivation_index_by_zipcode.r:
```diff
@@ -0,0 +1,25 @@
+
+# ZCTA data 
+ZCTA_data<- read.table("https://www2.census.gov/geo/docs/maps-data/data/rel/zcta_tract_rel_10.txt", header = TRUE,sep = ',',
+                       colClasses = c('ZCTA5' = 'character' , 'GEOID' = 'character'))
```
We need to only select `ZCTA5` and `GEOID` here so that the other variables do not get written into the data file (for example, `POPPT`, `HUPT`, etc)

cole-brokamp commented on deprivation_index_by_zipcode.r:
```diff
@@ -0,0 +1,25 @@
+library(tidyverse)
+
+# read deprivation index
+library(tidyverse)
```
don't need this line twice

cole-brokamp commented on dep_index.Rproj:
```diff
@@ -0,0 +1,13 @@
+Version: 1.0
```
remove the `.Rproj` file from the commit since its not part of the code required to make the dataset

cole-brokamp committed:
delete testing_file.txt

cole-brokamp committed:
Delete dep_index.Rproj

cole-brokamp committed:
simplify ZCTA calculation

- avoided writing extra variables included in the ZCTA crosswalk
- changed so variables in the global environment are never overwritten

cole-brokamp commented:
I think I've made all the changes necessary.  You just need to pull this down to your local repo, run and replace the CSV and RDS files based on the updated code and then push this back up.

cole-brokamp commented:
I almost forgot that we will need to also add a section to the README that the data is now also available at the ZCTA level.

cole-brokamp committed:
renamed ZCTA file to be consistent with repo structure

nobel1154 committed:
updated

nobel1154 committed:
update

Merge branch 'master' of https://github.com/nobel1154/dep_index

# Conflicts:
#	deprivation_index_by_zipcode.r

cole-brokamp commented:
closing this because I repushed it up after I reviewed it locally as #3. in the future, we should use branch names for pull requests so that its easier to pull/push and collaborate on them.
